spec/dynamic/layout.md

```diff
@@ -0,0 +1,333 @@
+|`*const impl Thin`[^2]|any[^1] | `<=size`[^1] | When initialized |
+|`&T`|Same as `*const T`| Same as `*const T`| Non-null, aligned to `T`[^4][^5]|
+|`&mut T`|Same as `*mut T`| Same as `*mut T`| Non-null, aligned to `T`[^4][^5]|
+|`Option<&T>`|Same as `*const T` | Same as `*const T`| aligned to `T`[^4][^5]|
+|`Option<&mut T>`|Same as `*mut T` | Same as `*mut T`| aligned to `T`[^4][^5]|
+|`fn()`[^3]| any[^1] | `<=ssize`[^1] | Non-null[^4] |
+| `Option<fn()>`[^3]| any[^1] | `<=size`[^1] | When initialized |
+| `[T;N]` | `size_of::<T>()*N` | Same as `T` | Each element valid |
+| `(T0,T1,..Tn)`[^6] | At least total of `T0`, `T1`, .. `Tn` + padding | At least maximum of `T0`, `T1`, ..`Tn` | Each element valid |
+| `()` | `0` | `1`  | Always |
+| `struct Foo{f0: T0, f1: T1, ..fN: Tn}`[^6][^7] | At least total of `T0`, `T1`, .. `Tn` + padding | At least maximum of `T0`, `T1`, ..`Tn` | Each field valid |
+| `struct Foo{}`[^7][^8] | `0` | `1` | Always |
+| `union Foo{f0: T0, f1: T1, ..fN; Tn}`[^6] | At least maximum of `T0`, `T1`, ..`Tn` | At least maximum of `T0`, `T1`, ..`Tn` |  |
+
+[^1]: Target defined
+
+[^2]: `usize` and pointers to `Thin` types have the same size and alignment requirement. Neither type has any padding bytes.
+
+[^3]: `fn()` refers to any fn-pointer type, regardless of return type, arguments, or ABI Tag
+
+[^4]: Range of valid values when computing as a suitably sized integer type (or pointer). Implicitly imposes a constraint of "When initialized"
+
+[^5]: References, and `Option`s of references (when `Some`) have additional aliasing constraints, see r#[dynamic.alias].
+
+[^6]: `repr(Rust)` types have an unspecified and unstable layout
+
+[^7]: Same as equivalent tuple struct. Struct size and alignment requirement may be modified by the `#[repr(align)]`
+
+[^8]: Same as equivalent unit struct
+
+## Layout Properties [§](r[dynamic.layout.properties])
+
+[§](r[dynamic.layout.properties.size]) Each `Sized` type has a property called it's size. The size is a non-negative integer.
+
+[!NOTE]: For example, the size of `u8` is `1`, and the size of `u32` is `4`.
+
+[§](r[dynamic.layout.properties.align]) Each `Sized` type and the slice type has a property call its alignment requirement.  The alignment requirement is a positive integer power of 2.
+
+[!NOTE]: The alignment requirement may simply be called the alignment or align of the type.
+
+[§](r[dynamic.layout.properties.size-align-value]) The size and alignment of any type is representible as a value of type `usize`. The alignment of a type divides its size.
+
+[!NOTE]: The minimum size of a type is `0`, and the minimum alignment requirement is `1`. A type that has size `0` and alignment requirement `1` is called a 1-ZST.
+
+[§](r[dynamic.layout.properties.storage]) Storage for a value is suitable for storing a value of a `Sized` type if it is at least the size of that type and the address of the start of the storage satisfies the alignment requirement.
```

Comment:
You forgot to define what it means to "satisfy the alignment requirement", the spec only ever said it was a pow2 number